File: app/src/main/java/com/locallinkonline/stcatherineschool/rest/tasks/StCatherineTask.java
package com.locallinkonline.stcatherineschool.rest.tasks;

import android.os.AsyncTask;

import com.google.gson.GsonBuilder;
import com.locallinkonline.stcatherineschool.rest.api.StCatherineApi;
import com.locallinkonline.stcatherineschool.room.db.StCatherineDatabase;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public abstract class StCatherineTask extends AsyncTask<Void, Void, Void> {

    final StCatherineApi stCatherineApi;

    public StCatherineTask(StCatherineDatabase db, String url) {
        this.stCatherineApi = new Retrofit.Builder()
                .baseUrl(url)
                .addConverterFactory(GsonConverterFactory.create(
                        new GsonBuilder().setLenient().create()))
                .build().create(StCatherineApi.class);
    }
}

File: app/src/main/java/com/locallinkonline/stcatherineschool/rest/tasks/StCatherineTask.kt
package com.locallinkonline.stcatherineschool.rest.tasks

import android.os.AsyncTask

import com.google.gson.GsonBuilder
import com.locallinkonline.stcatherineschool.rest.api.StCatherineApi
import com.locallinkonline.stcatherineschool.room.db.StCatherineDatabase

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

abstract class StCatherineTask<T>(db: StCatherineDatabase, url: String) : AsyncTask<T, Void, Void>() {

    internal val stCatherineApi: StCatherineApi = Retrofit.Builder()
            .baseUrl(url)
            .addConverterFactory(GsonConverterFactory.create(
                    GsonBuilder().setLenient().create()))
            .build().create(StCatherineApi::class.java)

}
